In [1]:
from dotenv import load_dotenv
load_dotenv()

True

In [2]:
documents = [
    "Machine learning is a subset of artificial intelligence",
    "Deep learning uses neural networks with multiple layers",
    "Python is a programming language popular for data science",
    "India won the cricket world cup in 2011",
    "AI models can understand and generate human language"
]

In [4]:
from langchain_google_genai import GoogleGenerativeAIEmbeddings

In [16]:
embeddings = GoogleGenerativeAIEmbeddings(model="gemini-embedding-2-preview")

In [17]:
vector = embeddings.embed_documents(documents)

In [18]:
len(vector[0])

3072

#Question by me -> 
Why do OpenAI and Gemini both produce 3072-dimensional embeddings?

The embedding dimension is simply the size of the output vector chosen by the model designers. Different embedding models can use the same dimensionality while learning completely different embedding spaces. Even if OpenAI and Gemini both output 3072-dimensional vectors, the values, training process, and semantic representations are different, so their embeddings are not directly interchangeable.